homeassistant/components/vizio/config_flow.py

```diff
@@ -336,26 +370,43 @@ def __init__(self) -> None:
             errors=errors,
         )
 
-    async def _pairing_complete(self, step_id: str) -> Dict[str, Any]:
-        """Handle config flow completion."""
+    async def async_step_pairing_complete_import(
+        self, user_input: Dict[str, Any] = None
+    ) -> Dict[str, Any]:
+        """Complete import config flow by displaying final message to show user access token and give further instructions."""
         if not self._must_show_form:
             return await self._create_entry_if_unique(self._data)
 
         self._must_show_form = False
         return self.async_show_form(
-            step_id=step_id,
+            step_id="pairing_complete_import",
             data_schema=vol.Schema({}),
             description_placeholders={"access_token": self._data[CONF_ACCESS_TOKEN]},
         )
 
-    async def async_step_pairing_complete(
+    async def async_step_tv_apps(
         self, user_input: Dict[str, Any] = None
     ) -> Dict[str, Any]:
-        """Complete non-import config flow by displaying final message to confirm pairing."""
-        return await self._pairing_complete("pairing_complete")
+        """Handle app configuration to complete TV configuration."""
+        if user_input is not None:
+            if user_input.get(CONF_APPS_TO_INCLUDE_OR_EXCLUDE):
+                # Update stored apps with user entry config keys
+                self._apps[user_input[CONF_INCLUDE_OR_EXCLUDE].lower()] = user_input[
+                    CONF_APPS_TO_INCLUDE_OR_EXCLUDE
+                ].copy()
 
-    async def async_step_pairing_complete_import(
-        self, user_input: Dict[str, Any] = None
-    ) -> Dict[str, Any]:
-        """Complete import config flow by displaying final message to show user access token and give further instructions."""
-        return await self._pairing_complete("pairing_complete_import")
+            return await self._create_entry_if_unique(self._data)
```

Comment:
We should do the unique check way sooner. Now a user is going through 2 pages of config just to get an abort message.

Reply:
good point, didn't think about that when I split the page into two